### ISSUE TITLE
False Positive: redundant_closure with macro

### ISSUE THREAD

**Issue by TyPR124** [rust-lang/rust-clippy#6732]
Lint name: redundant_closure

I tried this code:

```rust
use quote::quote;

fn _bar() {
    // Clippy suggests '||quote!()' => '$crate::__private::TokenStream::new'
    let _ = true.then(||quote!());
}
```

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cd500ed497d52778f4f2a0f171582f3b)

### Meta

`clippy 0.0.212 (cb75ad5d 2021-02-10)`

```
rustc 1.50.0 (cb75ad5db 2021-02-10)
binary: rustc
commit-hash: cb75ad5db02783e8b0222fee363c5f63f7e2cf5b
commit-date: 2021-02-10
host: x86_64-pc-windows-msvc
release: 1.50.0
```
